src/sentry/api/endpoints/artifact_lookup.py

```diff
@@ -0,0 +1,154 @@
+            try:
+                debug_ids.append(normalize_debug_id(debug_id))
+            except SymbolicError:
+                pass
+
+        # urls = request.GET.getlist("urls")
+        # release = request.GET.get("release")
+        # dist = request.GET.get("dist")
+
+        # For debug_ids, we will query for the artifact_bundle/file_id directly
+        bundle_file_ids = set(
+            DebugIdArtifactBundle.objects.filter(
+                organization_id=project.organization.id,
+                debug_id__in=debug_ids,
+            )
+            .select_related("artifact_bundle")
+            .values_list("artifact_bundle__file_id", flat=True)
+            .distinct("artifact_bundle__file_id")
+        )
+
+        # If we have `urls`, we want to:
+        # First, get the newest X artifact bundles, and *look inside them* to
+        # figure out if the file is included in them
+        # XXX: does that even work for partial matches?
+        # As in: `urls` would have `"/path/to/file"` for a file with the full url
+        # `"~/path/to/file.min.js"`.
+
+        # TODO: also query for and return bundles based on `urls`
+
+        # Possibly use the algorithm sketched up here:
+        # https://github.com/getsentry/sentry/pull/45697#issuecomment-1466389132
+        # That would narrow down our set of bundles to the minimum set that covers
+        # the file names we are querying for, and also leave us with the remaining
+        # set of file names that are not covered by any bundle, to look up below
+        # TODO: add those to `bundle_file_ids`
+
+        # Second, look for a legacy `ReleaseFile` (or whatever) if an individual
+        # file exists matching the release/dist/url we are looking for.
+
+        # TODO: also query for and return individual artifacts
+        individual_files = set()
+
+        # Then: Construct our response
+
+        found_artifacts = []
+        for file_id in bundle_file_ids:
+            # TODO: is there a better way to construct a url to this same route?
+            url = f"{base_url}?download={file_id}"
+            found_artifacts.append(
+                {
+                    "type": "bundle",
+                    "url": url,
+                }
+            )
+
+        for file in individual_files:
+            # TODO: is there a better way to construct a url to this same route?
+            url = f"{base_url}?download={file.id}"
+            found_artifacts.append(
+                {
+                    "type": "bundle",
+                    "url": url,
+                    # I believe `name` is the url/abs_path of the file?
+                    # As in: `"~/path/to/file.min.js"`?
+                    "abs_path": file.name,
+                    # These headers should ideally include the `Sourcemap` reference
+                    "headers": file.headers,
+                }
+            )
+
+        # TODO: how to properly paginate this thing?
```

Comment:
You can base it on release files endpoint https://github.com/getsentry/sentry/blob/6afde3f5f9110f0895e18637b6efc075d37dede5/src/sentry/api/endpoints/project_release_files.py#L100-L106

Reply:
hm, that works with `QuerySet`s internally. Though I want to query multiple things and work with them internally.
I guess I can put a hard limit on the queries internally, as we do not expect this to yield too many results ideally.